File: packages/kotlin-sdk/src/test/kotlin/com/etegram/sdk/ReferenceTest.kt
package com.etegram.sdk

import kotlin.test.Test
import kotlin.test.assertEquals
import kotlin.test.assertTrue

class ReferenceTest {
    @Test
    fun `generateTransactionReference uses ETG prefix and expected length`() {
        val reference = generateTransactionReference(20)
        assertTrue(reference.startsWith("ETG"))
        assertEquals(23, reference.length)
    }
}

File: packages/flutter-sdk/test/reference_test.dart
import 'package:etegram_flutter_sdk/etegram_flutter_sdk.dart';
import 'package:flutter_test/flutter_test.dart';

void main() {
  test('generateTransactionReference uses ETG prefix and default size', () {
    final String reference = generateTransactionReference();

    expect(reference.startsWith('ETG'), isTrue);
    expect(reference.length, 23);
  });

  test('generateTransactionReference respects custom length', () {
    final String reference = generateTransactionReference(10);

    expect(reference.length, 13);
  });
}
